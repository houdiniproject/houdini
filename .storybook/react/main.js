// License: LGPL-3.0-or-later
// Based on https://github.com/rails/webpacker/issues/1004#issuecomment-628377930
process.env.NODE_ENV = "development";
const railsWebpackEnv = require("../../config/webpack/environment");

module.exports = {
  stories: [
    "../../app/packs/**/*!(--html).stories.mdx",
    "../../app/packs/**/*!(--html).stories.[tj]s?(x)"
  ],
  addons: [
    "@storybook/addon-links",
    "@storybook/addon-essentials",
    'storybook-addon-intl'
  ],
  webpackFinal: (config) => {
    const result = {
    // do mutation to the config
    ...config,
    resolve: {
      ...config.resolve,
      ...railsWebpackEnv.config.resolve,
      modules: railsWebpackEnv.resolvedModules.map((i) => i.value),
    },
    module: {
      ...config.module,
      rules: railsWebpackEnv.loaders
        .filter((i) => !["nodeModules", //not sure why this is here
        "moduleCss" // this addresses issues with our webpack config for css not matching what storybook wants
      ].includes(i.key) )
        .map((i) => i.value),
    },
    plugins: [
      ...config.plugins,
      ...railsWebpackEnv.plugins.map((i) => i.value),
    ],
    
  };

  result.module.rules.filter((i) => i.test.test('.ttf')).forEach((i) => {
    i.use=['url-loader']
  })


  return result;
},
};
