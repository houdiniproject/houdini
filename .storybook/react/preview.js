// License: LGPL-3.0-or-later
import intlDecorate from '../../app/packs/components/tests/intl';
const jest = require('jest-mock');
window.jest = jest;

export const decorators = [intlDecorate()]