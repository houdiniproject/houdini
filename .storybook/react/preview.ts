// License: LGPL-3.0-or-later
import {decorators} from './preview_common';

require('../../app/javascript/components/tests/decorators/roboto')

export {decorators};
