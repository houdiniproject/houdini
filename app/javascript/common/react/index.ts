// License: LGPL-3.0-or-later

/**
 * Adds type safety to your default props.
 *
 * @example
 * // your component
 * function CustomComponent(props:{className:string, disabled?:boolean}) : JSX.Element {
 * 	// not show for
 * }
 *
 * CustomComponent.defaultProps = defaultProps(CustomComponent, {
 * 	className: 'another'
 * })
 *
 * @example <caption>Function properly typechecks</caption>
 * // use the same component as previous example
 * CustomComponent.defaultProps = defaultProps(CustomComponent, {
 * 	className: 'another',
 * 	unexpectedProp: 3 // <-- Typecheck will throw a compile error because you didn't expect this.
 * })
 *
 * @param {(props:TProperty) => JSX.Element} _component - the component you want to create default props for
 * @param {Partial<TProperty>} defaultProps - the defaultProps for your component
 * @returns defaultProps
 * */
export function defaultProps<TProperty>(_component:(props:TProperty) => JSX.Element, defaultProps:Partial<TProperty>):Partial<TProperty> {
	return defaultProps;
}