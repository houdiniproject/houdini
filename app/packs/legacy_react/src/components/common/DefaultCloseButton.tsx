import * as React from "react";
import { action, observable } from "mobx";
import { Transition } from "react-transition-group";
import { CloseButton } from "./svg/CloseButton";
import color from "color";
import { observer } from "mobx-react";
import ScreenReaderOnlyText from "./ScreenReaderOnlyText";

interface DefaultCloseButtonProps {
  onClick?: () => void
}

const mainColor = '#969696'
const darkenedColor = color(mainColor).darken(0.1).hex()
const defaultStyles = {
  foreground: {
    fill: mainColor
  },
  background: {
    stroke: mainColor,
    fill: '#FFFFFF'
  }
}

const states: { [state: string]: any } = {
  entering: {
    foreground: {
      fill: darkenedColor,
      transition: 'fill 250ms ease-in-out'
    },
    background: {
      stroke: darkenedColor,
      transition: 'stroke 250ms ease-in-out'
    }
  },
  entered: {
    foreground: {
      fill: darkenedColor,
    },
    background: {
      stroke: darkenedColor,
    }
  },
  exiting: {
    foreground: {
      fill: mainColor,
      transition: 'fill 250ms ease-in-out'
    },
    background: {
      stroke: mainColor,
      transition: 'stroke 250ms ease-in-out'
    }
  },
  exited: {
    foreground: {
      fill: mainColor,
    },
    background: {
      stroke: mainColor,
    }
  }
}

@observer
export class DefaultCloseButton extends React.Component<DefaultCloseButtonProps, {}> {
  @observable
  hovering: boolean

  @observable
  focusing: boolean

  @action.bound
  mouseEnter() {
    this.hovering = true;
  }

  @action.bound
  mouseLeave() {
    this.hovering = false;
  }

  @action.bound
  keyDown(event: React.KeyboardEvent<HTMLAnchorElement>) {
    if (event.key == 'Enter') {
      event.preventDefault();
      this.props.onClick();
    }
  }

  render() {
    return <Transition in={this.hovering} timeout={250}>
      {(hoverState) => {
        const backgroundStyle = {
          ...defaultStyles.background,
          ...((states[hoverState] && states[hoverState].background) || {})
        }

        const foregroundStyle =
        {
          ...defaultStyles.foreground,
          ...((states[hoverState] && states[hoverState].foreground) || {})
        }

        return <a onMouseEnter={this.mouseEnter} onMouseLeave={this.mouseLeave} onClick={this.props.onClick} onKeyDown={this.keyDown} tabIndex={0} className={'focusable_item'}>
          <CloseButton backgroundCircleStyle={backgroundStyle}
            foregroundCircleStyle={foregroundStyle}
          /><ScreenReaderOnlyText>Close modal</ScreenReaderOnlyText>
        </a>

      }
      }
    </Transition>

  }
}
