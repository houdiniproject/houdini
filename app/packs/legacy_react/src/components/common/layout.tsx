// License: LGPL-3.0-or-later
import * as React from 'react';
import {observer} from "mobx-react";
import * as _ from 'lodash'
import { ClassNameable } from './fields';

function arrayify<T>(items: T[]|T){
  return items instanceof Array ? items : [items]
}

type ClassNameableChildren = React.ReactElement<ClassNameable>[]|React.ReactElement<ClassNameable>

export const TwoColumnFields: React.StatelessComponent<{children:ClassNameableChildren}> = (props) => {
    const children = arrayify(props.children)
    return <Row>
        {
            children.map((i:React.ReactElement<ClassNameable>) => {
                let className = ""
                if (_.last(children) !== i){
                    className += " u-paddingRight--10"
                }
                if (i.props.className){
                    className += i.props.className
                }
                return <Column colSpan={6} breakSize={'sm'}>
                 {React.cloneElement(i, {className: className})}
                </Column>
        })}
    </Row>
}

TwoColumnFields.displayName = 'TwoColumnFields'

export const ThreeColumnFields: React.StatelessComponent<{children:ClassNameableChildren}> = (props) => {
  const children = arrayify(props.children)
    return <Row>
        {
          children.map((i:React.ReactElement<ClassNameable>) => {
                let className = ""
                if (_.last(children) !== i){
                    className += " u-paddingRight--10"
                }
                if (i.props.className){
                    className += i.props.className
                }
                return <Column colSpan={4} breakSize={'sm'}>
                    {React.cloneElement(i, {className: className})}
                </Column>
            })}
    </Row>
}

ThreeColumnFields.displayName = 'ThreeColumnFields'


export const Row: React.StatelessComponent<{}> = (props:{children:React.ReactElement<any>[]|React.ReactElement<any>}) => {
    return <div className="row">
        {props.children}
    </div>
}

Row.displayName = 'Row'

type ColumnBreakSize = 'xs'| 'sm'|'md'|'lg'
type ColumnSpan = 1|2|3|4|5|6|7|8|9|10|11|12

interface ColumnProps {
    children:React.ReactElement<any>
    colSpan:ColumnSpan,
    breakSize:ColumnBreakSize
}

export const Column: React.StatelessComponent<ColumnProps> = (props) => {
    let className = `col-${props.breakSize}-${props.colSpan} `
    props.children.props
    if (props.children.props.className){
        className += props.children.props['className']
    }

    return React.cloneElement(props.children, {className: className})
}

Column.displayName = 'Column'
