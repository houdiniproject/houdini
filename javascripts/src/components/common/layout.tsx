// License: LGPL-3.0-or-later
import * as React from 'react';
import {observer} from "mobx-react";
import * as _ from 'lodash'

export const TwoColumnFields = observer((props:{children:Array<React.ReactElement<any>>}) => {
    return <div className="clearfix">
        {
            _.take(props.children, 2).map((i:React.ReactElement<any>) => {
                let className = "col-left-6"
                if (_.last(props.children) !== i){
                    className += " u-paddingRight--10"
                }
                if (i.props['className']){
                    className += i.props['className']
                }

                return React.cloneElement(i, {wrapperClassName: className})
        })}
    </div>
})

export const ThreeColumnFields = observer((props:{children:React.ReactElement<any>[]}) => {
    return <div className="clearfix">
        {
          _.take(props.children, 3).map((i:React.ReactElement<any>) => {
                let className = "col-left-4"
                if (_.last(props.children) !== i){
                    className += " u-paddingRight--10"
                }
                if (i.props['className']){
                    className += i.props['className']
                }

                return React.cloneElement(i, {wrapperClassName: className})
            })}
    </div>
})