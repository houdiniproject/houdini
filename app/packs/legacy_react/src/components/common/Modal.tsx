// License: LGPL-3.0-or-later
import * as React from 'react';
import { observer } from 'mobx-react';
import AriaModal from 'react-aria-modal';
import { DefaultCloseButton } from './DefaultCloseButton';
import BootstrapWrapper from './BootstrapWrapper';
import { Row, Column } from './layout';

export interface ModalProps {
  onClose?: () => void // if you want your modal to close, this needs to set modalActive to false
  modalActive?: boolean
  titleText?: string
  focusDialog?: boolean
  dialogStyle?: any
  showCloseButton?: boolean
  childGenerator: () => any
}

class Modal extends React.Component<ModalProps, {}> {

  static defaultProps = {
    dialogStyle: { minWidth: '768px' },
    showCloseButton: true
  }

  render() {
    const modal = this.props.modalActive ? <AriaModal mounted={this.props.modalActive} titleText={this.props.titleText} focusDialog={this.props.focusDialog}
      onExit={this.props.onClose} dialogStyle={this.props.dialogStyle}>
      <BootstrapWrapper>
        <header className='modal-header' style={{
          position: 'relative',
          padding: '12px 10px 12px 20px'
        }}>
          <Row>
            <Column colSpan={11} breakSize={'xs'}>
              <h3 className='modal-header-title' style={{ margin: 0 }}>{this.props.titleText}</h3>
            </Column>
            {this.props.showCloseButton ?
              <Column colSpan={1} breakSize={'xs'}>
                <div style={{ textAlign: 'right' }}>
                  <DefaultCloseButton onClick={() => this.props.onClose()} />
                </div>
              </Column> : false
            }
          </Row>
        </header>
        <div className="modal-body">
          <div style={{ position: 'relative' }}>
            {this.props.childGenerator()}
          </div>
        </div>
      </BootstrapWrapper>
    </AriaModal> : false

    return modal
  }

}

export default observer(Modal)



