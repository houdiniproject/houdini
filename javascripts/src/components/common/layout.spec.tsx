// License: LGPL-3.0-or-later
import * as React from 'react';
import 'jest';
import Modal, { ModalProps } from './Modal'
import { shallow, mount, ReactWrapper } from "enzyme";
import toJson from "enzyme-to-json";
import { DefaultCloseButton } from './DefaultCloseButton';
import { Column, Row, ThreeColumnFields, TwoColumnFields } from './layout';

const SimpleCheckerComponent = (props: { className?: string }) => {
  return <div></div>
}

describe('layout', () => {
  describe('Column', () => {
    it('sets proper class without none passed in', () => {
      const column = shallow(<Column colSpan={12} breakSize={'sm'}>
        <SimpleCheckerComponent />
      </Column>)

      expect(column.find(SimpleCheckerComponent).props()['className']).toBe('col-sm-12 ')
    })

    it('sets proper class with one passed in', () => {
      const column = shallow(<Column colSpan={1} breakSize={'lg'}>
        <SimpleCheckerComponent className="another_class" />
      </Column>)

      expect(column.find(SimpleCheckerComponent).props()['className'])
        .toBe('col-lg-1 another_class')
    })

    it('sets proper display name', () => {
      const column = shallow(<Column colSpan={1} breakSize={'lg'}>
        <SimpleCheckerComponent className="another_class" />
      </Column>)
      expect(Column.displayName).toBe('Column')
    })
  })

  describe('Row', () => {
    it('renders single child properly', () => {
      const row = mount(<Row>
        <SimpleCheckerComponent/>
      </Row>)

      expect(toJson(row)).toMatchSnapshot()
    })


    it('renders multiple children properly', () => {
      const row = mount(<Row>
        <SimpleCheckerComponent/>
        <SimpleCheckerComponent/>
      </Row>)

      expect(toJson(row)).toMatchSnapshot()
    })

    it('has correct display name', () => {
      const row = mount(<Row>
        <SimpleCheckerComponent/>
      </Row>)

      expect(Row.displayName).toBe('Row')
    })
  })

  describe('ThreeColumnFields',() => {
    it('renders single child properly', () => {
      const fields = mount(<ThreeColumnFields>
          <SimpleCheckerComponent/>
        </ThreeColumnFields>)

      expect(toJson(fields)).toMatchSnapshot()
    })

    it('renders multiple children properly', () => {
      const fields = mount(<ThreeColumnFields>
          <SimpleCheckerComponent/>
          <SimpleCheckerComponent/>
          <SimpleCheckerComponent/>
        </ThreeColumnFields>)

      expect(toJson(fields)).toMatchSnapshot()
    })
  })

  describe('TwoColumnFields',() => {
    it('renders single child properly', () => {
      const fields = mount(<TwoColumnFields>
          <SimpleCheckerComponent/>
        </TwoColumnFields>)

      expect(toJson(fields)).toMatchSnapshot()
    })

    it('renders multiple children properly', () => {
      const fields = mount(<TwoColumnFields>
          <SimpleCheckerComponent/>
          <SimpleCheckerComponent/>
        </TwoColumnFields>)

      expect(toJson(fields)).toMatchSnapshot()
    })
  })
})