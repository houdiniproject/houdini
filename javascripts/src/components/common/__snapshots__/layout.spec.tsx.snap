// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout Row renders multiple children properly 1`] = `
<Row>
  <div
    className="row"
  >
    <SimpleCheckerComponent>
      <div />
    </SimpleCheckerComponent>
    <SimpleCheckerComponent>
      <div />
    </SimpleCheckerComponent>
  </div>
</Row>
`;

exports[`layout Row renders single child properly 1`] = `
<Row>
  <div
    className="row"
  >
    <SimpleCheckerComponent>
      <div />
    </SimpleCheckerComponent>
  </div>
</Row>
`;

exports[`layout ThreeColumnFields renders multiple children properly 1`] = `
<ThreeColumnFields>
  <Row>
    <div
      className="row"
    >
      <Column
        breakSize="sm"
        colSpan={4}
      >
        <SimpleCheckerComponent
          className="col-sm-4  u-paddingRight--10"
        >
          <div />
        </SimpleCheckerComponent>
      </Column>
      <Column
        breakSize="sm"
        colSpan={4}
      >
        <SimpleCheckerComponent
          className="col-sm-4  u-paddingRight--10"
        >
          <div />
        </SimpleCheckerComponent>
      </Column>
      <Column
        breakSize="sm"
        colSpan={4}
      >
        <SimpleCheckerComponent
          className="col-sm-4 "
        >
          <div />
        </SimpleCheckerComponent>
      </Column>
    </div>
  </Row>
</ThreeColumnFields>
`;

exports[`layout ThreeColumnFields renders single child properly 1`] = `
<ThreeColumnFields>
  <Row>
    <div
      className="row"
    >
      <Column
        breakSize="sm"
        colSpan={4}
      >
        <SimpleCheckerComponent
          className="col-sm-4 "
        >
          <div />
        </SimpleCheckerComponent>
      </Column>
    </div>
  </Row>
</ThreeColumnFields>
`;

exports[`layout TwoColumnFields renders multiple children properly 1`] = `
<TwoColumnFields>
  <Row>
    <div
      className="row"
    >
      <Column
        breakSize="sm"
        colSpan={6}
      >
        <SimpleCheckerComponent
          className="col-sm-6  u-paddingRight--10"
        >
          <div />
        </SimpleCheckerComponent>
      </Column>
      <Column
        breakSize="sm"
        colSpan={6}
      >
        <SimpleCheckerComponent
          className="col-sm-6 "
        >
          <div />
        </SimpleCheckerComponent>
      </Column>
    </div>
  </Row>
</TwoColumnFields>
`;

exports[`layout TwoColumnFields renders single child properly 1`] = `
<TwoColumnFields>
  <Row>
    <div
      className="row"
    >
      <Column
        breakSize="sm"
        colSpan={6}
      >
        <SimpleCheckerComponent
          className="col-sm-6 "
        >
          <div />
        </SimpleCheckerComponent>
      </Column>
    </div>
  </Row>
</TwoColumnFields>
`;
